pkg/detectors/paytm/paytm.go

```diff
@@ -0,0 +1,92 @@
+package paytm
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"paytm"}) + `\b([0-9a-zA-Z]{20})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"paytm"}
+}
+
+// FromData will find and optionally verify Paytm secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+
+		s1 := detectors.Result{
+			DetectorType: detectorspb.DetectorType_Paytm,
+			Raw:          []byte(resMatch),
+		}
+
+		if verify {
+			client := s.client
+			if client == nil {
+				client = defaultClient
+			}
+
+			payload := strings.NewReader(`{"requestType":"Payment","mid":"Mid-1234","websiteName":"Sample-website-name","orderId":"ORDERID_1234","callbackUrl":"http://localhost:3000/paytm","txnAmount":{"value":"1.00","currenty":"INR"},"userInfo":{"custId":"customer-id-12345"}}`)
```

Comment:
@fumblehool anyway we could have this not hit `https://securegw.paytm.in/theia/api/v1/initiateTransaction?`? Seems like there is a risk of sending funds